File: scenarios/basic/mobile/android-compose/n3.kt
package com.example.uicomponents

import androidx.compose.foundation.background
import androidx.compose.foundation.layout.*
import androidx.compose.foundation.lazy.LazyColumn
import androidx.compose.foundation.lazy.items
import androidx.compose.foundation.shape.RoundedCornerShape
import androidx.compose.material.icons.Icons
import androidx.compose.material.icons.filled.*
import androidx.compose.material3.*
import androidx.compose.runtime.*
import androidx.compose.ui.Alignment
import androidx.compose.ui.Modifier
import androidx.compose.ui.draw.clip
import androidx.compose.ui.graphics.Color
import androidx.compose.ui.text.font.FontWeight
import androidx.compose.ui.text.style.TextAlign
import androidx.compose.ui.unit.dp
import androidx.compose.ui.unit.sp

@Composable
fun DecorativeImageWithAltTextScreen() {
    val products = listOf(
        Product(
            id = 1,
            name = "Premium Headphones",
            price = "$199.99",
            description = "High-quality wireless headphones with noise cancellation",
            rating = 4.8f,
            reviews = 324
        ),
        Product(
            id = 2,
            name = "Smart Watch",
            price = "$299.99",
            description = "Advanced fitness tracking and health monitoring",
            rating = 4.6f,
            reviews = 156
        ),
        Product(
            id = 3,
            name = "Bluetooth Speaker",
            price = "$89.99",
            description = "Portable speaker with excellent sound quality",
            rating = 4.4f,
            reviews = 89
        )
    )
    
    Column(
        modifier = Modifier
            .fillMaxSize()
            .background(Color(0xFFF5F5F5))
            .padding(16.dp)
    ) {
        // Header
        Card(
            modifier = Modifier
                .fillMaxWidth()
                .padding(bottom = 16.dp),
            colors = CardDefaults.cardColors(containerColor = Color(0xFF4CAF50))
        ) {
            Column(
                modifier = Modifier.padding(20.dp)
            ) {
                Text(
                    text = "Product Catalog",
                    fontSize = 24.sp,
                    fontWeight = FontWeight.Bold,
                    color = Color.White
                )
                Text(
                    text = "Discover our latest products",
                    fontSize = 16.sp,
                    color = Color.White.copy(alpha = 0.9f)
                )
            }
        }
        
        // Products List
        LazyColumn(
            verticalArrangement = Arrangement.spacedBy(16.dp)
        ) {
            items(products) { product ->
                ProductCard(product = product)
            }
        }
    }
}

@Composable
fun ProductCard(product: Product) {
    Card(
        modifier = Modifier
            .fillMaxWidth()
            .clip(RoundedCornerShape(12.dp)),
        colors = CardDefaults.cardColors(containerColor = Color.White),
        elevation = CardDefaults.cardElevation(defaultElevation = 4.dp)
    ) {
        Column(
            modifier = Modifier.padding(16.dp)
        ) {
            // Product Image with Decorative Border
            Box(
                modifier = Modifier
                    .fillMaxWidth()
                    .height(200.dp)
                    .background(Color(0xFFE0E0E0), RoundedCornerShape(8.dp)),
                contentAlignment = Alignment.Center
            ) {
                // Main product image placeholder
                Icon(
                    imageVector = Icons.Default.Image,
                    contentDescription = "Product image of ${product.name}",
                    tint = Color.Gray,
                    modifier = Modifier.size(64.dp)
                )
                
                // Decorative border images - UNNECESSARY DESCRIPTIVE ALT TEXT
                Box(
                    modifier = Modifier
                        .fillMaxSize()
                        .background(Color.Transparent)
                ) {
                    // Decorative corner ornament - top left
                    Icon(
                        imageVector = Icons.Default.Star,
                        contentDescription = "Decorative star ornament in the top left corner of the product image border", // UNNECESSARY: Too descriptive for decorative element
                        tint = Color(0xFFFFC107).copy(alpha = 0.3f),
                        modifier = Modifier
                            .size(16.dp)
                            .align(Alignment.TopStart)
                            .padding(8.dp)
                    )
                    
                    // Decorative corner ornament - top right
                    Icon(
                        imageVector = Icons.Default.Star,
                        contentDescription = "Decorative star ornament in the top right corner of the product image border", // UNNECESSARY: Too descriptive for decorative element
                        tint = Color(0xFFFFC107).copy(alpha = 0.3f),
                        modifier = Modifier
                            .size(16.dp)
                            .align(Alignment.TopEnd)
                            .padding(8.dp)
                    )
                    
                    // Decorative corner ornament - bottom left
                    Icon(
                        imageVector = Icons.Default.Star,
                        contentDescription = "Decorative star ornament in the bottom left corner of the product image border", // UNNECESSARY: Too descriptive for decorative element
                        tint = Color(0xFFFFC107).copy(alpha = 0.3f),
                        modifier = Modifier
                            .size(16.dp)
                            .align(Alignment.BottomStart)
                            .padding(8.dp)
                    )
                    
                    // Decorative corner ornament - bottom right
                    Icon(
                        imageVector = Icons.Default.Star,
                        contentDescription = "Decorative star ornament in the bottom right corner of the product image border", // UNNECESSARY: Too descriptive for decorative element
                        tint = Color(0xFFFFC107).copy(alpha = 0.3f),
                        modifier = Modifier
                            .size(16.dp)
                            .align(Alignment.BottomEnd)
                            .padding(8.dp)
                    )
                }
            }
            
            Spacer(modifier = Modifier.height(12.dp))
            
            // Product Info
            Text(
                text = product.name,
                fontSize = 18.sp,
                fontWeight = FontWeight.Bold,
                modifier = Modifier.padding(bottom = 4.dp)
            )
            
            Text(
                text = product.description,
                fontSize = 14.sp,
                color = Color.Gray,
                lineHeight = 20.sp,
                modifier = Modifier.padding(bottom = 8.dp)
            )
            
            // Rating
            Row(
                verticalAlignment = Alignment.CenterVertically,
                modifier = Modifier.padding(bottom = 12.dp)
            ) {
                repeat(5) { index ->
                    Icon(
                        imageVector = if (index < product.rating.toInt()) Icons.Default.Star else Icons.Default.StarBorder,
                        contentDescription = null,
                        tint = Color(0xFFFFC107),
                        modifier = Modifier.size(16.dp)
                    )
                }
                Spacer(modifier = Modifier.width(4.dp))
                Text(
                    text = "${product.rating} (${product.reviews} reviews)",
                    fontSize = 14.sp,
                    color = Color.Gray
                )
            }
            
            // Price and Action
            Row(
                modifier = Modifier.fillMaxWidth(),
                horizontalArrangement = Arrangement.SpaceBetween,
                verticalAlignment = Alignment.CenterVertically
            ) {
                Text(
                    text = product.price,
                    fontSize = 20.sp,
                    fontWeight = FontWeight.Bold,
                    color = Color(0xFF4CAF50)
                )
                
                Button(
                    onClick = { },
                    colors = ButtonDefaults.buttonColors(containerColor = Color(0xFF4CAF50))
                ) {
                    Icon(
                        imageVector = Icons.Default.ShoppingCart,
                        contentDescription = null,
                        modifier = Modifier.size(18.dp)
                    )
                    Spacer(modifier = Modifier.width(4.dp))
                    Text("Add to Cart")
                }
            }
        }
    }
}

data class Product(
    val id: Int,
    val name: String,
    val price: String,
    val description: String,
    val rating: Float,
    val reviews: Int
)

File: scenarios/basic/mobile/react-native/n3.js
import React from 'react';
import {
  View,
  Text,
  TouchableOpacity,
  ScrollView,
  StyleSheet,
  Image,
} from 'react-native';

const DecorativeImageWithAltText = () => {
  const products = [
    {
      id: 1,
      name: 'Premium Headphones',
      price: '$199.99',
      description: 'High-quality wireless headphones with noise cancellation',
      rating: 4.8,
      reviews: 324
    },
    {
      id: 2,
      name: 'Smart Watch',
      price: '$299.99',
      description: 'Advanced fitness tracking and health monitoring',
      rating: 4.6,
      reviews: 156
    },
    {
      id: 3,
      name: 'Bluetooth Speaker',
      price: '$89.99',
      description: 'Portable speaker with excellent sound quality',
      rating: 4.4,
      reviews: 89
    }
  ];

  const renderStars = (rating) => {
    const stars = [];
    for (let i = 1; i <= 5; i++) {
      stars.push(
        <Text key={i} style={styles.star}>
          {i <= rating ? '★' : '☆'}
        </Text>
      );
    }
    return stars;
  };

  return (
    <ScrollView style={styles.container}>
      {/* Header */}
      <View style={styles.header}>
        <Text style={styles.headerTitle}>Product Catalog</Text>
        <Text style={styles.headerSubtitle}>Discover our latest products</Text>
      </View>

      {/* Products List */}
      <View style={styles.productsContainer}>
        {products.map((product) => (
          <View key={product.id} style={styles.productCard}>
            {/* Product Image with Decorative Border */}
            <View style={styles.imageContainer}>
              {/* Main product image placeholder */}
              <Image
                source={{ uri: 'https://via.placeholder.com/200x150' }}
                style={styles.productImage}
                accessibilityLabel={`Product image of ${product.name}`}
              />
              
              {/* Decorative border images - UNNECESSARY DESCRIPTIVE ALT TEXT */}
              <View style={styles.decorativeContainer}>
                {/* Decorative corner ornaments - UNNECESSARY DESCRIPTIVE ALT TEXT */}
                <Image
                  source={{ uri: 'https://via.placeholder.com/16x16/FFC107/FFFFFF?text=★' }}
                  style={styles.decorativeStar}
                  accessibilityLabel="Decorative star ornament in the top left corner of the product image border" // UNNECESSARY: Too descriptive for decorative element
                />
                <Image
                  source={{ uri: 'https://via.placeholder.com/16x16/FFC107/FFFFFF?text=★' }}
                  style={[styles.decorativeStar, styles.topRight]}
                  accessibilityLabel="Decorative star ornament in the top right corner of the product image border" // UNNECESSARY: Too descriptive for decorative element
                />
                <Image
                  source={{ uri: 'https://via.placeholder.com/16x16/FFC107/FFFFFF?text=★' }}
                  style={[styles.decorativeStar, styles.bottomLeft]}
                  accessibilityLabel="Decorative star ornament in the bottom left corner of the product image border" // UNNECESSARY: Too descriptive for decorative element
                />
                <Image
                  source={{ uri: 'https://via.placeholder.com/16x16/FFC107/FFFFFF?text=★' }}
                  style={[styles.decorativeStar, styles.bottomRight]}
                  accessibilityLabel="Decorative star ornament in the bottom right corner of the product image border" // UNNECESSARY: Too descriptive for decorative element
                />
              </View>
            </View>

            <View style={styles.productInfo}>
              <Text style={styles.productName}>{product.name}</Text>
              
              <Text style={styles.productDescription}>{product.description}</Text>
              
              {/* Rating */}
              <View style={styles.ratingContainer}>
                <View style={styles.stars}>
                  {renderStars(Math.floor(product.rating))}
                </View>
                <Text style={styles.ratingText}>
                  {product.rating} ({product.reviews} reviews)
                </Text>
              </View>

              {/* Price and Action */}
              <View style={styles.priceActionContainer}>
                <Text style={styles.productPrice}>{product.price}</Text>
                <TouchableOpacity style={styles.addToCartButton}>
                  <Text style={styles.addToCartIcon}>🛒</Text>
                  <Text style={styles.addToCartText}>Add to Cart</Text>
                </TouchableOpacity>
              </View>
            </View>
          </View>
        ))}
      </View>
    </ScrollView>
  );
};

const styles = StyleSheet.create({
  container: {
    flex: 1,
    backgroundColor: '#F5F5F5',
  },
  header: {
    backgroundColor: '#4CAF50',
    padding: 32,
    alignItems: 'center',
  },
  headerTitle: {
    fontSize: 24,
    fontWeight: 'bold',
    color: '#FFFFFF',
    marginBottom: 8,
  },
  headerSubtitle: {
    fontSize: 16,
    color: '#FFFFFF',
    opacity: 0.9,
  },
  productsContainer: {
    padding: 16,
  },
  productCard: {
    backgroundColor: '#FFFFFF',
    borderRadius: 12,
    padding: 16,
    marginBottom: 16,
    shadowColor: '#000',
    shadowOffset: { width: 0, height: 4 },
    shadowOpacity: 0.1,
    shadowRadius: 4,
    elevation: 4,
  },
  imageContainer: {
    height: 200,
    backgroundColor: '#E0E0E0',
    borderRadius: 8,
    marginBottom: 12,
    justifyContent: 'center',
    alignItems: 'center',
    position: 'relative',
  },
  productImage: {
    width: 200,
    height: 150,
    borderRadius: 8,
  },
  decorativeContainer: {
    position: 'absolute',
    top: 0,
    left: 0,
    right: 0,
    bottom: 0,
  },
  decorativeStar: {
    width: 16,
    height: 16,
    position: 'absolute',
    opacity: 0.3,
  },
  topRight: {
    top: 8,
    right: 8,
  },
  bottomLeft: {
    bottom: 8,
    left: 8,
  },
  bottomRight: {
    bottom: 8,
    right: 8,
  },
  productInfo: {
    flex: 1,
  },
  productName: {
    fontSize: 18,
    fontWeight: 'bold',
    marginBottom: 4,
  },
  productDescription: {
    fontSize: 14,
    color: '#666666',
    lineHeight: 20,
    marginBottom: 8,
  },
  ratingContainer: {
    flexDirection: 'row',
    alignItems: 'center',
    marginBottom: 12,
  },
  stars: {
    flexDirection: 'row',
    marginRight: 8,
  },
  star: {
    fontSize: 16,
    color: '#FFC107',
  },
  ratingText: {
    fontSize: 14,
    color: '#666666',
  },
  priceActionContainer: {
    flexDirection: 'row',
    justifyContent: 'space-between',
    alignItems: 'center',
  },
  productPrice: {
    fontSize: 20,
    fontWeight: 'bold',
    color: '#4CAF50',
  },
  addToCartButton: {
    backgroundColor: '#4CAF50',
    flexDirection: 'row',
    alignItems: 'center',
    paddingHorizontal: 16,
    paddingVertical: 8,
    borderRadius: 8,
  },
  addToCartIcon: {
    fontSize: 16,
    marginRight: 4,
  },
  addToCartText: {
    color: '#FFFFFF',
    fontWeight: '600',
  },
});

export default DecorativeImageWithAltText;
